src/graph/util/test/ExpressionUtilsTest.cpp

```diff
@@ -764,5 +764,52 @@ TEST_F(ExpressionUtilsTest, expandExpression) {
     ASSERT_EQ(expected, target->toString());
   }
 }
+
+TEST_F(ExpressionUtilsTest, simplifyLogicalExpr) {
+  {
+    auto filter = parse("A and true");
+    auto target =
+        ExpressionUtils::simplifyLogicalExpr(static_cast<const LogicalExpression *>(filter));
+    auto expected = "A";
+    ASSERT_EQ(expected, target->toString());
+  }
+  {
+    auto filter = parse("A and false");
+    auto target =
+        ExpressionUtils::simplifyLogicalExpr(static_cast<const LogicalExpression *>(filter));
+    auto expected = "false";
+    ASSERT_EQ(expected, target->toString());
+  }
+  {
+    auto filter = parse("A or true");
+    auto target =
+        ExpressionUtils::simplifyLogicalExpr(static_cast<const LogicalExpression *>(filter));
+    auto expected = "true";
+    ASSERT_EQ(expected, target->toString());
+  }
+  {
+    auto filter = parse("A or false");
+    auto target =
+        ExpressionUtils::simplifyLogicalExpr(static_cast<const LogicalExpression *>(filter));
+    auto expected = "A";
+    ASSERT_EQ(expected, target->toString());
+  }
+  {
+    auto filter = parse("A or 2 > 1");
+    auto target =
+        ExpressionUtils::simplifyLogicalExpr(static_cast<const LogicalExpression *>(filter));
+    auto expected = "(A OR (2>1))";
```

Comment:
why is this expression not true?

Reply:
we should fold the `2>1` relational expression to constant `true` expr firstly and then continue to fold the logical expression

Reply:
Yes, but this case is just to test `simplifyLogicalExpr` not `foldConstant`.
When we actually use it, we should first apply FoldConstant, and then apply simplifyLogicalExpr.

Reply:
A good way to apply these optimizations for expressions is to apply the optimization rules to the plan in the optimizer.
It should looks like the following:
```
Optimizer::Optimizer(Binder &binder, ClientContext &context) : context(context), binder(binder), rewriter(context) {
	rewriter.rules.push_back(make_unique<ConstantFoldingRule>(rewriter));
	rewriter.rules.push_back(make_unique<DistributivityRule>(rewriter));
	rewriter.rules.push_back(make_unique<ArithmeticSimplificationRule>(rewriter));
	rewriter.rules.push_back(make_unique<CaseSimplificationRule>(rewriter));
	rewriter.rules.push_back(make_unique<ConjunctionSimplificationRule>(rewriter));
}
```

Reply:
ok.